File: demo_laya_js/src/ui/LetoExtendedDemo.js
export default class LetoExtendedDemo extends Laya.Scene {
    constructor() {
        super();
        
        //加载场景文件
        this.loadScene("test/LetoExtendedDemo.scene");

        // event
        this.btnBack.on(Laya.Event.CLICK, this, this.onBackClicked)
        this.btnDefault.on(Laya.Event.CLICK, this, this.onDefaultClicked)
        this.btnCustomLogic.on(Laya.Event.CLICK, this, this.onCustomLogicClicked)
        this.btnNormalClaim.on(Laya.Event.CLICK, this, this.onNormalClaimClicked)
        this.btnNoVideo.on(Laya.Event.CLICK, this, this.onNoVideoClicked)
    }

    onAwake() {
        LTExtendedJSSDK.setAdListener(this)
    }

    adId() {
        return 1
    }

    onDefaultClicked() {
        LTExtendedJSSDK.load(this.adId());
        LTExtendedJSSDK.show(this.adId(), {
            coin: 1,
            ratio: 2
        })
    } 

    onCustomLogicClicked() {
        LTExtendedJSSDK.load(this.adId(), {
            style: {
                title: `自定义标题`,
                video_button_title: '自定义按钮文字',
                icon: 'http://download.mgc-games.com/access/upload/20190319/5c909dc73468a.png'
            }
        })
        LTExtendedJSSDK.show(this.adId(), {
            coin: 1,
            ratio: 2,
            custom_logic: true
        })
    }

    onNormalClaimClicked() {
        LTExtendedJSSDK.load(this.adId(), {
            style: {
                title: '带普通领取按钮',
                show_my_coin: true,
                show_normal_button: true
            }
        })
        LTExtendedJSSDK.show(this.adId(), {
            coin: 1,
            ratio: 2
        })
    }

    onNoVideoClicked() {
        LTExtendedJSSDK.load(this.adId(), {
            style: {
                title: '比例<=1则无视频, 只能普通领取',
                show_my_coin: true
            }
        })
        LTExtendedJSSDK.show(this.adId(), {
            coin: 1,
            ratio: 1
        })
    }

    onBackClicked() {
        Laya.Scene.open('test/LetoMainDemo.scene')
    }

    onExtendedAdLoaded(adId) {
        LTJSSDK.printLog("LetoExtendDemo::onExtendedAdLoaded(" + adId + ")");
    }

    onExtendedAdFailed(adId, errorInfo) {
        LTJSSDK.printLog("LetoExtendDemo::onExtendedAdFailed(" + adId + ", " + errorInfo + ")");
    }

    onExtendedAdShow(adId) {
        LTJSSDK.printLog("LetoExtendDemo::onExtendedAdShow(" + adId + ")");
    }

    onExtendedAdClose(adId, res) {
        LTJSSDK.printLog("LetoExtendDemo::onExtendedAdClose(" + adId + ", " + JSON.stringify(res) + ")");
    }

    onExtendedAdCustomClose(adId, res) {
        LTJSSDK.printLog("LetoExtendDemo::onExtendedAdCustomClose(" + adId + ", " + JSON.stringify(res) + ")");
    }

    onExtendedAdVideoClose(adId, res) {
        LTJSSDK.printLog("LetoExtendDemo::onExtendedAdVideoClose(" + adId + ", " + JSON.stringify(res) + ")");
    }
    
    onExtendedAdNormalClaim(adId) {
        LTJSSDK.printLog("LetoExtendDemo::onExtendedAdNormalClaim(" + adId + ")");
    }
}

File: demo_unity/Assets/Scripts/LetoExtendedDemo.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.SceneManagement;
using LetoAd;

public class LetoExtendedDemo : MonoBehaviour
{
    public ILTExtendedCSSDK _letoSdk;
    private int _adId = 1;

    void Awake() {
        _letoSdk = new LTExtendedCSSDK();
        _letoSdk.setAdListener(new EventListener(_letoSdk));
    }

    public void OnBackClicked() {
        SceneManager.LoadScene("LetoMainDemo");
    }

    public void OnShowDefaultClicked() {
        _letoSdk.load(_adId);
        LTExtendedShowParams p = new LTExtendedShowParams();
        p.coin = 1;
        p.ratio = 2;
        _letoSdk.show(_adId, p);
    }

    public void OnShowCustomLogicClicked() {
        LTExtendedLoadParams lp = new LTExtendedLoadParams();
        lp.style.title = "自定义标题";
        lp.style.video_button_title = "自定义按钮文字";
        lp.style.icon = "http://download.mgc-games.com/access/upload/20190319/5c909dc73468a.png";
        _letoSdk.load(_adId, lp);
        LTExtendedShowParams p = new LTExtendedShowParams();
        p.coin = 1;
        p.ratio = 2;
        p.custom_logic = true;
        _letoSdk.show(_adId, p);
    }

    public void OnShowNormalClaimClicked() {
        LTExtendedLoadParams lp = new LTExtendedLoadParams();
        lp.style.title = "带普通领取按钮";
        lp.style.show_my_coin = true;
        lp.style.show_normal_button = true;
        _letoSdk.load(_adId, lp);
        LTExtendedShowParams p = new LTExtendedShowParams();
        p.coin = 1;
        p.ratio = 2;
        _letoSdk.show(_adId, p);
    }

    public void OnShowNoVideoClicked() {
        LTExtendedLoadParams lp = new LTExtendedLoadParams();
        lp.style.title = "比例<=1则无视频, 只能普通领取";
        lp.style.show_my_coin = true;
        _letoSdk.load(_adId, lp);
        LTExtendedShowParams p = new LTExtendedShowParams();
        p.coin = 1;
        p.ratio = 1;
        _letoSdk.show(_adId, p);
    }

    class EventListener : ILTExtendedListener {
        private ILTExtendedCSSDK _letoSdk;

        public EventListener(ILTExtendedCSSDK letoSdk) {
            _letoSdk = letoSdk;
        }

        public void onExtendedLoaded(int adId) {
            Debug.Log("onExtendedLoaded: " + adId);
        }

        public void onExtendedFailed(int adId, string errMsg) {
            Debug.Log("onExtendedFailed: " + adId + ", errmsg: " + errMsg);
        }

        public void onExtendedShow(int adId) {
            Debug.Log("onExtendedShow: " + adId);
        }

        public void onExtendedClose(int adId, string res) {
            Debug.Log("onExtendedClose: " + adId + ", res: " + res);
            _letoSdk.destroy(adId);
        }

        public void onExtendedCustomClose(int adId, string res) {
            Debug.Log("onExtendedCustomClose: " + adId + ", res: " + res);
        }

        public void onExtendedVideoClose(int adId, string res) {
            Debug.Log("onExtendedVideoClose: " + adId + ", res: " + res);
        }

        public void onExtendedNormalClaim(int adId) {
            Debug.Log("onExtendedNormalClaim: " + adId);
        }
    }
}
